framework/web/SessionIterator.php

```
@@ -62,7 +63,7 @@ public function key()
     #[\ReturnTypeWillChange]
     public function current()
     {
-        return isset($_SESSION[$this->_key]) ? $_SESSION[$this->_key] : null;
+        return isset($_SESSION, $_SESSION[$this->_key]) ? $_SESSION[$this->_key] : null;
```

There is no point of checking `isset($_SESSION)` here.

---

@bizley it's fool protection, if somebody use iterator without starting session

---

Isn't it better to fail in this case rather than returning nothing?

---

@samdark it's declared behavior

---

Yes, but if `$_SESSION[$this->_key]`  is not set, it doesn't mater if `$_SESSION` is for these checks.